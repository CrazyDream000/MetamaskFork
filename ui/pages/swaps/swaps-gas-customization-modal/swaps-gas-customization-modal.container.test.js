import React from 'react';
import configureMockStore from 'redux-mock-store';

import {
  renderWithProvider,
  createSwapsMockStore,
} from '../../../../test/jest';
import GasCustomizationModalContainer from '.';

describe('GasCustomizationModalContainer', () => {
  it('renders the component with initial props', () => {
    const store = configureMockStore()(createSwapsMockStore());
    const { getByText, getByTestId } = renderWithProvider(
      <GasCustomizationModalContainer />,
      store,
    );
    expect(getByTestId('page-container__header')).toMatchSnapshot();
    expect(getByText('Basic')).toBeInTheDocument();
    expect(getByText('Advanced')).toBeInTheDocument();
    expect(getByText('Estimated Processing Times')).toBeInTheDocument();
    expect(getByText('Send Amount')).toBeInTheDocument();
    expect(getByText('Transaction Fee')).toBeInTheDocument();
    expect(
      getByTestId('gas-modal-content__info-row__send-info'),
    ).toMatchSnapshot();
    expect(
      getByTestId('gas-modal-content__info-row__transaction-info'),
    ).toMatchSnapshot();
    expect(
      getByTestId('gas-modal-content__info-row__total-info'),
    ).toMatchSnapshot();
    expect(getByText('Save')).toBeInTheDocument();
  });
});
