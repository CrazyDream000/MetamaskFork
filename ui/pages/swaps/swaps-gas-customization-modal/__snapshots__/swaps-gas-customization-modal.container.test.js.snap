// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GasCustomizationModalContainer renders the component with initial props 1`] = `
<div
  class="page-container__header page-container__header--no-padding-bottom"
  data-testid="page-container__header"
>
  <div
    class="page-container__title"
  >
    Customize Gas
  </div>
  <div
    class="page-container__subtitle"
  >
    Increasing fee may decrease processing times, but it is not guaranteed.
  </div>
  <a
    class="button btn-link page-container__header-close-text"
    role="button"
    tabindex="0"
  >
    Close
  </a>
  <ul
    class="page-container__tabs"
  >
    <li
      class="tab page-container__tab tab--active"
    >
      <button>
        Basic
      </button>
    </li>
    <li
      class="tab page-container__tab"
    >
      <button>
        Advanced
      </button>
    </li>
  </ul>
</div>
`;

exports[`GasCustomizationModalContainer renders the component with initial props 2`] = `
<div
  class="gas-modal-content__info-row__send-info"
  data-testid="gas-modal-content__info-row__send-info"
>
  <span
    class="gas-modal-content__info-row__send-info__label"
  >
    Send Amount
  </span>
  <span
    class="gas-modal-content__info-row__send-info__value"
  >
    0 ETH
  </span>
</div>
`;

exports[`GasCustomizationModalContainer renders the component with initial props 3`] = `
<div
  class="gas-modal-content__info-row__transaction-info"
  data-testid="gas-modal-content__info-row__transaction-info"
>
  <span
    class="gas-modal-content__info-row__transaction-info__label"
  >
    Transaction Fee
  </span>
  <span
    class="gas-modal-content__info-row__transaction-info__value"
  >
    0 ETH
  </span>
</div>
`;

exports[`GasCustomizationModalContainer renders the component with initial props 4`] = `
<div
  class="gas-modal-content__info-row__total-info"
  data-testid="gas-modal-content__info-row__total-info"
>
  <span
    class="gas-modal-content__info-row__total-info__label"
  >
    New Total
  </span>
  <span
    class="gas-modal-content__info-row__total-info__value"
  >
    0 ETH
  </span>
</div>
`;
